pkg/uninstall/uninstall.go

```diff
@@ -172,14 +212,25 @@ func (u *Uninstall) Run(ctx context.Context) error { //nolint:funlen
 		},
 	})
 
-	// There are no resources with finalizers in the monitoring namespace, so
-	// we can delete it directly
-	uninstallSteps = append(uninstallSteps, common.Step{
-		Desc: "Delete Operator Lifecycle Manager",
-		F: func(ctx context.Context) error {
-			return u.deleteOLM(ctx)
-		},
-	})
+	if u.config.SkipOLM {
+		u.l.Debug("Skipping OLM uninstallation")
+		uninstallSteps = append(uninstallSteps, common.Step{
+			Desc: "Delete Everest Catalog",
+			F: func(ctx context.Context) error {
+				u.l.Info("Deleting Everest catalog")
+				return u.kubeClient.DeleteCatalogSource(ctx, u.config.CatalogNamespace, catalogSource)
+			},
+		})
+	} else {
+		// There are no resources with finalizers in the monitoring namespace, so
+		// we can delete it directly
```

Comment:
I know this comment was already there, but it doesn't really make any sense. Maybe it was placed there by mistake during a previous  refactor?

Reply:
updated